packages/vscode-extension/src/project/deviceSession.ts

```diff
@@ -355,6 +361,29 @@ export class DeviceSession
         this.emitStateChange();
       }
     });
+    devtools.onEvent("appOrientationChanged", (orientation: AppOrientationType) => {
+      const isLandscape =
+        this.rotation === DeviceRotationType.LandscapeLeft ||
+        this.rotation === DeviceRotationType.LandscapeRight;
+
+      // if the app orientation is equal to "Landscape", it means we do not have enuogh
+      // information on the application site to infer the detailed orientation. 
+      // "Landscape" will only be messaged upon the initialisation - after that the app can infer the orientation
+      // based on our assumptions.
+      // If the device is in landscape mode, we assume that the app orientation is the same as the device rotation.
+      // Otherwise, it means the app is initialised and does not have previous orientation - 
```

Comment:
```suggestion
      // If the device is in landscape mode, we assume that the app orientation is the same as the device rotation.
      // Otherwise, we pick `LandscapeLeft`
```
I think this comment should say _why_ we pick `LandscapeLeft`, not detail the reasons why `"Landscape"` orientation was sent.
The reason, as I understand it, is that the application is forced into starting in landscape mode while the device itself was in portrait, and in that case `LandscapeLeft` seems to be the default, so I'd write something along the lines of

```suggestion
      // If the device is in landscape mode, we assume that the app orientation is the same as the device rotation.
      // Otherwise, we assume the application was started in Landscape mode while the device was oriented in Portrait, and we pick `LandscapeLeft` as the default orientation in that case.
```

Reply:
Also, I think it might be nicer to move these comments inside the `if` cases below, so they're directly above the `this.appOrientation = ` assignments they relate to.

Reply:
Moved to `if` statements in 78820032

```ts
devtools.onEvent("appOrientationChanged", (orientation: AppOrientationType) => {
  const isLandscape =
    this.rotation === DeviceRotationType.LandscapeLeft ||
    this.rotation === DeviceRotationType.LandscapeRight;

  if (orientation === "Landscape") {
    // if the app orientation is equal to "Landscape", it means we do not have enough
    // information on the application side to infer the detailed orientation.
    if (isLandscape) {
      // if the device is in landscape mode, we assume that the app orientation is correct with device rotation
      this.appOrientation = this.rotation;
    } else {
      // if the device is not in landscape mode we set app orientation to the last known orientation.
      // if the last orientation is not known, we assume the application was started in Landscape mode
      // while the device was oriented in Portrait, and we pick `LandscapeLeft` as the default orientation in that case.
      this.appOrientation = this.appOrientation ?? DeviceRotationType.LandscapeLeft;
    }
  } else {
    this.appOrientation = orientation;
  }

  this.emitStateChange();
});
return devtools;
```